src/operator/leaky_relu-inl.h

```diff
@@ -177,9 +182,20 @@ class LeakyReLUOp : public Operator {
       case leakyrelu::kPReLU: {
         weight = in_data[leakyrelu::kGamma].get<xpu, 1, real_t>(s);
         grad_weight = in_grad[leakyrelu::kGamma].get<xpu, 1, real_t>(s);
-        grad_weight = sumall_except_dim<1>(F<prelu_grad>(data) * grad);
-        gdata = F<mshadow_op::xelu_grad>(data, mshadow::expr::broadcast<1>(weight, data.shape_))
-                * grad;
+        if (weight.shape_[0] == 1) {
```

Comment:
Could you explain? I thought prelu was already supported

Reply:
Since uyou are adding those I would suggest also adding a learnable ISRLU:
https://arxiv.org/abs/1710.09967

Reply:
Existing PReLU had two problems:
1. gamma parameter was never documented and can't be passed in using kwargs.
2. it doesn't support scalar broadcast as was attempted in #8912

Reply:
If you meant to test for scalar than you should use `weight.shape_.Size()`

Reply:
It's equivalent because weight is `Tensor<xpu, 1>`. I think `Size()` is safer choice in case `weight` changes definition.

Reply:
Infershape sets it to data.shape[1].
When would weight's shape be (1,)?

Reply:
If the weight parameter is shared across all axis, then the only one scalar value is shared everywhere, in which case the weight should be (1,)

Reply:
But infershape doesn't allow this

Reply:
@bradcar sorry that I missed your comment earlier, and thanks for sharing your work. In this PR I'd like to first focus on wrapping up the previous two PRs for activations. Since you wrote the paper, would you like to implement that in mxnet?

Reply:
There are two options, either writing it in Gluon by defining hybrid_forward in python, or extending the leaky relu operator in C for better performance.